schema/deploy/roles/idle-session-disconnector/grants.sql

```diff
@@ -0,0 +1,33 @@
+-- Deploy seattleflu/id3c-customizations:roles/idle-session-disconnector/grants to pg
+
+begin;
+
+revoke all on database :"DBNAME" from "idle-session-disconnector";
+revoke all on schema receiving, warehouse, shipping from "idle-session-disconnector";
+revoke all on all tables in schema receiving, warehouse, shipping from "idle-session-disconnector";
+
+grant connect on database :"DBNAME" to "idle-session-disconnector";
+
+-- Regular users cannot see the state of sessions that they don't own. Create a
+-- new view using security definer with limited columns that this role can
+-- access to see a session's state.
+create or replace function get_stat_activity() returns table(
+    pid integer, usename name, application_name text, client_addr inet,
+    state_change timestamp with time zone, state text) as
+    $$
+    select
+        pid, usename, application_name, client_addr, state_change, state
+        from pg_catalog.pg_stat_activity;
+    $$
+    language sql
+    volatile
+    security definer
+    set search_path = pg_catalog;
+
+create view pg_stat_activity_nonsuperuser as
+    select * from get_stat_activity();
```

Comment:
Similarly, would be good to explicitly name the schema the view should be created in.

I would also make the function and view names parallel the ones used by the `pg_stat_activity` view and its `pg_stat_get_activity` function, so renaming our function to `pg_stat_get_activity_nonsuperuser`.

Reply:
Renamed